### DISCUSSION TITLE
No information about selected resources or applied filters in action fields() when selecting "all matching" resources

### DISCUSSION THREAD

**Discussion by mattias-sanfridsson** [laravel/nova-issues#5360] [Ideas & Feature Requests]
- Laravel Version: 9.52.4
- Nova Version: 4.22.1
- PHP Version: 8.1.14
- Database Driver & Version: MySQL 8.0
- Operating System and Version: MacOS Ventura 13.2.1
- Browser type and version: Chrome 110.0.5481.100
- Reproduction Repository: -

### Description:
Selecting all when having a filter applied will only give the request the following data in an action's `field` method: 
```
{"relationshipType":null,"display":"index","resources":"all"}
```
Whereas selecting specific resources will give the resource IDs: 
```
{"relationshipType":null,"display":"index","resources":["33545","33544"]}
```
Since there's no detail about what filter has been applied it's impossible to filter the action field data based on the selected resources. `$request->selectedResources()` will also return `null`.

The filter I'm using is a default BelongsTo: 
```php
BelongsTo::make(__('Company'), 'company', Company::class)->filterable(),
```

And the code I'm using to log the data in the action's `field` method looks like this:
```php
public function fields(NovaRequest $request): array
{
    info('request', $request->all());

    return [];
}
```

### Detailed steps to reproduce the issue on a fresh Nova installation:
- Set up a resource with a belongsto field that is filterable.
- Set up an action that logs the incoming request data in the fields() method.
- Go to the list, apply the filter and select all matching.
- Compare the log with selecting specific resources and opening the action.

**Comment by crynobone** [comment]
![image](https://user-images.githubusercontent.com/172966/221802214-f5371cbc-fe35-4ce7-be2f-d76b9f3052a0.png)

The filters information should be available from `$request->query()`.

**Reply by mattias-sanfridsson** [reply]
@crynobone thanks, but that gives me the same result as previously stated: 
```
public function fields(NovaRequest $request): array
{
    info('Request', $request->query());

    return [];
}
```
```
Request {"relationshipType":null,"display":"index","resources":"all"}
```

**Reply by crynobone** [reply]
> Since there's no detail about what filter has been applied it's impossible to filter the action field data based on the selected resources. $request->selectedResources() will also return null.

Okay, but still `$request->selectedResources()` will always return `null` when the value is `all`. This is to avoid Action selection on millions of rows and causing applications to hang/be unresponsive.

**Reply by mattias-sanfridsson** [reply]
@crynobone But wouldn't it make sense to at least include the applied filter somehow? Right now there's a risk that a change would be applied on _all_ resources, not the filtered ones because there's no information about the filter in the request.

**Reply by crynobone** [reply]
Filtered information is not being used by Nova on those requests, so it isn't useful for us. Also, the value is only useful when the value is mapped to the query (not for this request).

**Reply by mattias-sanfridsson** [reply]
But if you apply a filter you can select all matching resources and then run an action on them?

**Reply by crynobone** [reply]
> But if you apply a filter you can select all matching resources and then run an action on them?

Action selection and Running actions are two different requests.

**Reply by mattias-sanfridsson** [reply]
Ok so I'll draw the conclusion that it's impossible to know what resources are filtered in the fields method then :)
